shared/nm-utils/nm-secret-utils.c

```diff
@@ -132,3 +133,31 @@ nm_secret_buf_to_gbytes_take (NMSecretBuf *secret, gssize actual_len)
 	                                   _secret_buf_free,
 	                                   secret);
 }
+
+/*****************************************************************************/
+
+/**
+ * nm_utils_memeqzero_secret:
+ * @data: the data pointer to check (may be %NULL if @length is zero).
+ * @length: the number of bytes to check.
+ *
+ * Checks that all bytes are zero. This always takes the same amount
+ * of time to prevent timing attacks.
+ *
+ * Returns: whether all bytes are zero.
+ */
+gboolean
+nm_utils_memeqzero_secret (gconstpointer data, gsize length)
+{
+	const guint8 *const key = data;
+	volatile guint8 acc = 0;
+	gsize i;
+
+	/* Based on key_is_zero() by Jason A. Donenfeld <Jason@zx2c4.com>. */
```

Comment:
You really don't this now.

Reply:
The comment is not for Copyright reasons, but to understand where the code originates from. For what it's worth, 2 lines of the functions are non-obvious to me, and it seems helpful to be able to find the original implementation, to possibly understand why these lines are the way they are.

Anyway, it would be hard to argue that any comment is ever really "needed". Dropped.